Add background image and css parallax effect

mammuth opened the pull request:
ToDo:
- Browser testing

Future ToDo:
- ~Option for real parallax via JS~

Related issue #10 

![background-image](https://cloud.githubusercontent.com/assets/3121306/25860562/304939fc-34e2-11e7-83f9-d605d6a76f88.png)

mammuth committed:
Add background image and css parallax effect
Related issue #10

SteinRobert commented:
Do you think we should enable to user to set the size of the background image via easy-thumbnails or something? I think this might be something we should consider in order to enable the editors to minimize the images delivered to the website's visitors.

mammuth commented:
> Do you think we should enable to user to set the size of the background image via easy-thumbnails or something?

Yes absolutely! However, I'd suggest presets like "small (512 width)", ..., "large (1980 width)" or something like that. This way the user does not have to think much about it.

mammuth committed:
Allow the user to specify the maximum size of the background image
 The resized image will be generated using easy-thumbnails. The default size is set to 1080p to avoid serving large, original-sized images.

mammuth commented on layouter/models.py:
```diff
@@ -62,13 +64,40 @@ class ContainerPlugin(CMSPlugin):
+        except:
+            return ''
+        else:
+            return image.url
```
Should we do other / better error handling here? There are a couple of exceptions easy-thumbnails might throw.

mammuth committed:
Add true, js-based parallax effect and form validation

SteinRobert commented on layouter/templates/layouter/container.html:
```diff
@@ -2,14 +2,38 @@
     </div>
 </div>
+
+{% if instance.background_image_parallax == instance.TRUE_PARALLAX %}
```
This is nice. When I read about the feature, this was actually the first thing that came to my mind - only use the .js if needed. (y)

SteinRobert commented on layouter/models.py:
```diff
@@ -58,17 +61,62 @@ class ContainerPlugin(CMSPlugin):
+    CSS_PARALLAX = 1
+    TRUE_PARALLAX = 2
+    BACKGROUND_IMAGE_PARALLAX_CHOICES = (
+        (NO_PARALLAX, _('No Effect')),
```
Maybe we can find a better name for it? Something like "fixed background image" - and "parallax background image".

SteinRobert commented on layouter/cms_plugins.py:
```diff
@@ -19,6 +19,12 @@ class ContainerCMSPlugin(CMSPluginBase):
         (None, {
             'fields': ('container_type', 'margin')
         }),
+        (_('Background'), {
```
I like it. I'm thinking about organizing the admin in tabs, not in accordions in the future - this could make the overview a bit more clear. But not for now.

SteinRobert reviewed (changes requested):
Just a naming thing for the parallax choices - maybe there is a more non-technical name for the choices we have there.

mammuth committed:
Rename parallax effect options
"Fixed background image" and "Parallax background image"

mammuth committed:
Rebase onto master and adapt new template layout

mammuth committed:
Add migration for background options
